From 2aee160e209cbdcc44a31b99fffb77e273ebaf0b Mon Sep 17 00:00:00 2001
From: lihuang <putin.li@rock-chips.com>
Date: Fri, 20 Apr 2018 16:09:40 +0800
Subject: [PATCH] MALI: bifrost: RK: There are a few modifications in some
 'Kbuild' and 'Kconfig' files.

A new config MALI_BIFROST_FOR_LINUX is introduced to specify which directory of Birfrost to use.

Change-Id: I1f4a01e963073c68238b08f930b196aa799b8e17
Signed-off-by: Putin Lee <putin.li@rock-chips.com>
---
 drivers/gpu/arm/Kbuild                    |  4 +++-
 drivers/gpu/arm/Kconfig                   | 11 +++++++++++
 drivers/gpu/arm/bifrost_for_linux/Kconfig |  3 +--
 3 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/arm/Kbuild b/drivers/gpu/arm/Kbuild
index 30246dc556f5..5fdf2044b785 100644
--- a/drivers/gpu/arm/Kbuild
+++ b/drivers/gpu/arm/Kbuild
@@ -26,4 +26,6 @@ obj-$(CONFIG_MALI_MIDGARD_FOR_ANDROID) += midgard/
 
 obj-$(CONFIG_MALI400)      += mali400/
 
-obj-$(CONFIG_MALI_BIFROST) += bifrost/
+obj-$(CONFIG_MALI_BIFROST_FOR_ANDROID) += bifrost/
+
+obj-$(CONFIG_MALI_BIFROST_FOR_LINUX) += bifrost_for_linux/
diff --git a/drivers/gpu/arm/Kconfig b/drivers/gpu/arm/Kconfig
index 8403a1a48c00..599711c18af9 100644
--- a/drivers/gpu/arm/Kconfig
+++ b/drivers/gpu/arm/Kconfig
@@ -34,4 +34,15 @@ endchoice
 
 source "drivers/gpu/arm/midgard/Kconfig"
 
+choice
+	prompt "Mali bifrost driver"
+
+config MALI_BIFROST_FOR_ANDROID
+	bool "Mali Bifrost for Android"
+
+config MALI_BIFROST_FOR_LINUX
+	bool "Mali Bifrost for Linux only"
+
+endchoice
+
 source "drivers/gpu/arm/bifrost/Kconfig"
diff --git a/drivers/gpu/arm/bifrost_for_linux/Kconfig b/drivers/gpu/arm/bifrost_for_linux/Kconfig
index 7c70914e3900..e78b634031ee 100644
--- a/drivers/gpu/arm/bifrost_for_linux/Kconfig
+++ b/drivers/gpu/arm/bifrost_for_linux/Kconfig
@@ -15,8 +15,7 @@
 
 
 menuconfig MALI_BIFROST
-	tristate "Mali Bifrost series support"
-	select GPU_TRACEPOINTS if ANDROID
+	tristate "Mali Bifrost series support (Linux only)"
 	default n
 	help
 	  Enable this option to build support for a ARM Mali Bifrost GPU.
-- 
2.35.3

